heartbeat/monitors/active/dialchain/tls.go

```diff
@@ -35,24 +35,39 @@ import (
 //        "rtt": { "handshake": { "us": ... }}
 //    }
 //  }
-func TLSLayer(cfg *transport.TLSConfig, to time.Duration) Layer {
+func TLSLayer(cfg *transport.TLSConfig, to time.Duration, meta *transport.DialerMeta) Layer {
 	return func(event common.MapStr, next transport.Dialer) (transport.Dialer, error) {
 		var timer timer
 
 		// Wrap next dialer so to start the timer when 'next' returns.
 		// This gets us the timestamp for when the TLS layer will start the handshake.
 		next = startTimerAfterDial(&timer, next)
 
-		dialer, err := transport.TLSDialer(next, cfg, to)
+		dialer, err := transport.TLSMetaDialer(next, cfg, to, meta)
 		if err != nil {
 			return nil, err
 		}
 
 		return afterDial(dialer, func(conn net.Conn) (net.Conn, error) {
 			// TODO: extract TLS connection parameters from connection object.
-
 			timer.stop()
 			event.Put("tls.rtt.handshake", look.RTT(timer.duration()))
+
+			connState := meta.Tls.ConnectionState()
+			certs := make([]common.MapStr, len(connState.VerifiedChains))
+			for chainIndex, chain := range connState.VerifiedChains {
+				for certIndex, cert := range chain {
+					j := common.MapStr{
+						"chain_index":      chainIndex,
```

Comment:
This is only a subset of the fields for now, but I would expand this to the full field list.

Reply:
Also take into account what @ph contributed to ECS: https://github.com/elastic/ecs#tls Perhaps we can extend this further.